packages/server/src/auth/domain/RegisterAsUserHandler.ts

```diff
@@ -33,7 +33,9 @@ class RegisterAsUserHandler implements Handler<RegisterAsUserRequest, void> {
 
     await authDoc.save();
     // TODO pass non static token
-    await this.authMailService.sendUserConfirmation(email, 'dcmDkmygBGIILTWaDeP0GFIljjf19R2C');
+    if (process.env.NODE_ENV !== 'test') {
```

Comment:
Ten if ma dwa problemy:

1. jest nieskalowalny - dodając w innym miejscu wysyłanie maila trzeba znowu dodać if-a i tak w kółko,
2. kod aplikacji nie powinien wiedzieć o tym, że jest testowany, bo jakby nie było to aplikacja działa bez testów.

Te problemy można [w bardzo łatwy sposób rozwiązać korzystając z Dependency Injection](https://docs.nestjs.com/fundamentals/testing#end-to-end-testing). Wystarczy w testach podmienić service odpowiedzialny za wysyłanie maila na jakiś jego stub, np.
```ts
class MailerServiceStub {
  sendMail(sendMailOptions: ISendMailOptions) {
    return Promise.resolve({ status: 200, options: sendMailOptions });
  }
}
```
podmiana DI dla testów [tutaj](https://github.com/CodersCamp2021-HK/CodersCamp2021.Project.Fullstack/blob/main/packages/server/test/shared/E2eFixture.ts#L30-L33):
```ts
    const moduleFixture: TestingModule = await override(
      Test.createTestingModule({
        imports: [AppModule],
        controllers: options.testControllers,
      })
        .overrideProvider(MailerService)
        .useClass(MailerServiceStub),
    ).compile();
```

P.S. w JS można to też rozwiązać w inny sposób, ze względu na dynamiczny system importu modułów, można w testach użyć `jest.mock('../module')`, ale to rozwiązanie nie skaluje się do innych języków programowania więc lepiej korzystać z DI.